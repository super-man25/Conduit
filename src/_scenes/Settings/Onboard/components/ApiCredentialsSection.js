import React from 'react';

import { H3, FormInput } from '_components';
import { InputGroup, InputContainer } from './styled';

export const ApiCredentialsSection = () => (
  <>
    <H3 type="tertiary">API Credentials</H3>
    <InputGroup>
      <InputContainer>
        <FormInput
          label="Account ID"
          name="accountId"
          validate={(value) => {
            if (!value) return 'Account ID is required';
            if (isNaN(value)) return 'Account ID must be a number';
          }}
        />
      </InputContainer>
      <InputContainer>
        <FormInput
          label="Api Key"
          name="apiKey"
          validate={(value) => !value && 'Api key is required'}
        />
      </InputContainer>
      <InputContainer>
        <FormInput
          label="Default Vendor ID"
          name="defaultVendorId"
          validate={(value) => {
            if (!value) return 'Default vendor ID is required';
            if (isNaN(value)) return 'Default vendor ID must be a number';
          }}
        />
      </InputContainer>
    </InputGroup>
  </>
);
