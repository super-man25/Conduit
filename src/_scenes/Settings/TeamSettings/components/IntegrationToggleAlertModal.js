// @flow
import * as React from 'react';
import styled, { keyframes } from 'styled-components';
import { cssConstants } from '_constants';
import {
  H4,
  Box,
  Text,
  Flex,
  SecondaryButton,
  PrimaryButton
} from '_components';
import type { EDIntegration } from '_models';

export const slideUpAnimation = keyframes`
  0% {
    opacity: 0;
    transform: translate(-50%, -20%);
  }
  100% {
    opacity: 1;
  }
`;

export const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  opacity: 0.3;
  background-color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

const ModalContent = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 500px;
  z-index: 9;
  box-shadow: 0 1px 10px rgba(0, 0, 0, 0.27);
  animation: ${slideUpAnimation} 200ms ease-out;
  animation-fill-mode: forwards;
`;

const ModalHeader = styled.div`
  padding: 1.5rem;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};
  border-bottom: 1px solid ${cssConstants.PRIMARY_DARK_GRAY};
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
`;

const ModalBody = styled.div`
  padding: 1.5rem;
  background-color: ${cssConstants.PRIMARY_WHITE};
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 3px;
`;

export const Title = H4.extend`
  margin: 0;
  margin-bottom: 0.25rem;
  color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

type Props = {
  onCancel: () => void,
  onConfirm: () => void,
  integration: EDIntegration
};

export class IntegrationToggleAlertModal extends React.Component<Props> {
  componentDidMount() {
    document.addEventListener('keydown', this.closeOnEscapePressed);
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.closeOnEscapePressed);
  }

  closeOnEscapePressed = (e: KeyboardEvent) => {
    if (e.key === 'Escape') {
      this.props.onCancel();
    }
  };

  render() {
    const { integration, onCancel, onConfirm } = this.props;

    return (
      <Box>
        <ModalContent>
          <ModalHeader>
            <Title>Toggle Secondary Integration?</Title>
          </ModalHeader>
          <ModalBody>
            <Box marginBottom="2rem">
              <Text size={16} color={cssConstants.PRIMARY_DARKEST_GRAY}>
                This action will turn the{' '}
                <span style={{ color: cssConstants.SECONDARY_BLUE }}>
                  {integration.name}{' '}
                </span>
                secondary integration{' '}
                <span style={{ color: cssConstants.SECONDARY_BLUE }}>
                  {integration.isActive ? 'off' : 'on'}
                </span>{' '}
                for all events. Are you sure you would like to continue?
              </Text>
            </Box>
            <Flex justify="flex-end">
              <SecondaryButton small onClick={onCancel} margin="0 1rem 0 0">
                Cancel
              </SecondaryButton>
              <PrimaryButton small onClick={onConfirm}>
                Continue
              </PrimaryButton>
            </Flex>
          </ModalBody>
        </ModalContent>
        <ModalOverlay onClick={onCancel} />
      </Box>
    );
  }
}
