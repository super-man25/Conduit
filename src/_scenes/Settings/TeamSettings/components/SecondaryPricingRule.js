import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import {
  SettingEditButton,
  PrimaryButton,
  NumberInputField,
  Flex,
  FlexItem,
  Input
} from '_components';
import { withClickAway } from '_hoc';
import { actions } from '_state/client';

const ClickAwayDiv = withClickAway(Fragment);

export class SecondaryPricingRulePresenter extends Component {
  state = {
    isEditing: false,
    id: this.props.id,
    percent: this.props.percent,
    constant: this.props.constant,
    percentInvalid: false,
    constantInvalid: false
  };

  reset = () => {
    this.setState({
      isEditing: false,
      id: this.props.id,
      percent: this.props.percent,
      constant: this.props.constant,
      percentInvalid: false,
      constantInvalid: false
    });
  };

  toggleEdit = () => {
    this.setState({ isEditing: !this.state.isEditing });
  };

  saveSettings = () => {
    const { id, percent, constant } = this.state;
    const payload = {
      id,
      percent: percent !== '' ? percent : null,
      constant: constant !== '' ? constant : null,
      onSuccess: this.handleSuccess,
      onError: this.handleError
    };
    this.props.updateSecondaryPricingRule(payload);
  };

  handleSuccess = () => {
    this.setState({
      percentInvalid: false,
      constantInvalid: false
    });
    this.toggleEdit();
  };

  handleError = (error) => {
    error.path === 'percent'
      ? this.setState({ isEditing: true, percentInvalid: true })
      : this.setState({ isEditing: true, constantInvalid: true });
  };

  update = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value,
      percentInvalid: false,
      constantInvalid: false
    });
  };

  handleFocus = (name) => {
    name === 'percent'
      ? this.setState({ constant: null })
      : this.setState({ percent: null });
  };

  render() {
    return (
      <ClickAwayDiv onClickAway={this.reset}>
        {this.state.isEditing ? (
          <Fragment>
            <Flex inline direction="row" align="center" marginBottom="15px">
              <FlexItem margin="0 10px 0 0">
                <NumberInputField
                  component={Input}
                  name="percent"
                  value={this.state.percent || ''}
                  onChange={(e) => this.update(e)}
                  onFocus={(e) => this.handleFocus(e.target.name)}
                  placeholder="Percent"
                  inValid={this.state.percentInvalid}
                />
              </FlexItem>
              <FlexItem margin="0 0 0 10px">
                <NumberInputField
                  component={Input}
                  name="constant"
                  value={this.state.constant || ''}
                  onChange={(e) => this.update(e)}
                  onFocus={(e) => this.handleFocus(e.target.name)}
                  placeholder="Constant"
                  inValid={this.state.constantInvalid}
                />
              </FlexItem>
            </Flex>
            <PrimaryButton onClick={this.saveSettings}>SAVE</PrimaryButton>
          </Fragment>
        ) : (
          <SettingEditButton weight="bold" onClick={this.toggleEdit} />
        )}
      </ClickAwayDiv>
    );
  }
}

const mapDispatchToProps = {
  updateSecondaryPricingRule: actions.updateSecondaryPricingRule
};

export const SecondaryPricingRule = connect(
  null,
  mapDispatchToProps
)(SecondaryPricingRulePresenter);
