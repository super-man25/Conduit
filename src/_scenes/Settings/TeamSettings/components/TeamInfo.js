// @flow

import {
  Setting,
  H5,
  Flex,
  EDText,
  SelectBox,
  SettingBorderButton
} from '_components';
import { toggleEditing } from '../stateChanges';
import React from 'react';
import { titleCase, orDash } from '_helpers/string-utils';
import { withClickAway } from '_hoc';

const TeamInfoWrapper = Flex.extend`
  flex-direction: column;
  justify-content: left;
  margin: 0;
  width: 85%;
`;

const SettingButtonGroup = Flex.extend`
  justify-content: space-evenly;
  align-items: center;
`;

const TeamSetting = withClickAway(Setting.extend``);

type Props = {
  name: string,
  pricingInterval: number,
  dirtyPricingInterval: number,
  resetDirtyPricingInterval: () => void,
  setPricingInterval: (number) => void,
  update: ({ pricingInterval: number }) => void
};

type State = {
  isEditing: boolean
};

type Interval = {
  label: string,
  value: number
};

const intervals: Array<Interval> = [
  { label: '15 minutes', value: 15 },
  { label: '30 minutes', value: 30 },
  { label: '1 hour', value: 60 },
  { label: '6 hours', value: 360 },
  { label: 'Daily', value: 1440 }
];

export class TeamInfo extends React.Component<Props, State> {
  state = {
    isEditing: false
  };

  toggleEditButton = (event: SyntheticMouseEvent<HTMLElement>) => {
    this.setState(toggleEditing);
    this.props.resetDirtyPricingInterval();
  };

  clickAway = () => {
    this.setState({ isEditing: false });
  };

  handlePricingIntervalChange = (
    event: SyntheticInputEvent<HTMLSelectElement>
  ) => {
    const { value } = event.target;
    const pricingInterval = parseInt(value, 10);
    this.props.setPricingInterval(pricingInterval);
  };

  handlePricingIntervalSave = (event: SyntheticMouseEvent<HTMLElement>) => {
    const { dirtyPricingInterval } = this.props;
    this.props.update({ pricingInterval: dirtyPricingInterval });
    this.setState(toggleEditing);
  };

  get pricingDescription() {
    const { pricingInterval } = this.props;

    if (!pricingInterval) {
      return `--`;
    }

    const interval = intervals.find(
      (interval) => interval.value === pricingInterval
    );

    if (interval) {
      return titleCase(interval.label);
    }

    return titleCase(`${pricingInterval} minutes`);
  }

  render() {
    const { isEditing } = this.state;
    const { name, dirtyPricingInterval } = this.props;

    const textProps = {
      size: 'large',
      margin: '1rem 0'
    };

    return (
      <TeamInfoWrapper>
        <H5 type="secondary">Basic Settings</H5>
        <Setting cols={3}>
          <EDText {...textProps}>Team</EDText>
          <EDText {...textProps}>{titleCase(orDash(name))}</EDText>
          <EDText {...textProps} />
        </Setting>
        <TeamSetting cols={3} onClickAway={this.clickAway}>
          <EDText {...textProps}>Generate pricing updates every</EDText>
          {isEditing ? (
            <React.Fragment>
              <SelectBox
                noBg
                size="large"
                onChange={this.handlePricingIntervalChange}
                value={dirtyPricingInterval}
              >
                {intervals.map((o, key) => (
                  <option key={key} value={o.value}>
                    {o.label}
                  </option>
                ))}
              </SelectBox>
              <SettingButtonGroup>
                <SettingBorderButton onClick={this.handlePricingIntervalSave}>
                  Save
                </SettingBorderButton>
              </SettingButtonGroup>
            </React.Fragment>
          ) : (
            <React.Fragment>
              <EDText {...textProps}>{this.pricingDescription}</EDText>
            </React.Fragment>
          )}
        </TeamSetting>
      </TeamInfoWrapper>
    );
  }
}

export default TeamInfo;
