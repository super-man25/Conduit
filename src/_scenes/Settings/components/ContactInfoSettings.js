import {
  HelpBlockDiv,
  Input,
  LinkButton,
  Setting,
  SettingCancelButton,
  SettingEditableValue,
  SettingReadonlyValue,
  SettingSaveButton,
  Spacing
} from '_components';
import { cssConstants } from '_constants';
import { validateEmail, validatePhoneNumber } from '_helpers/string-utils';
import { actions as userActions } from '_state/auth';
import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import styled from 'styled-components';

const ContactInfoContainer = styled.div`
  width: 100%;
  height: auto;
  margin: 0;
  margin-bottom: 20px;
  padding: 0;
  color: ${cssConstants.PRIMARY_GRAY};
  font-size: 13px;
  font-weight: 300;
  letter-spacing: 0.4px;
  > * {
    color: ${cssConstants.PRIMARY_DARKEST_GRAY};
    font-size: 15px;
    font-weight: 400;
    letter-spacing: 0.4px;
    line-height: 30px;
  }
  background: transparent;
`;

class ContactInfoSettings extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      confirmPassword: '',
      editEmail: false,
      editName: false,
      editPassword: false,
      editPhoneNumber: false,
      newPassword: '',
      password: '',
      passwordsMatch: false,
      user: props.authState.model,
      touchedFields: {}
    };

    this.handleBlur = this.handleBlur.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleEditClick = this.handleEditClick.bind(this);
    this.handlePasswordChange = this.handlePasswordChange.bind(this);
    this.matchingPasswordCheck = this.matchingPasswordCheck.bind(this);
    this.reset = this.reset.bind(this);
    this.saveEmail = this.saveEmail.bind(this);
    this.savePersonalInfo = this.savePersonalInfo.bind(this);
    this.savePassword = this.savePassword.bind(this);
  }

  reset() {
    this.setState({
      confirmPassword: '',
      editEmail: false,
      editName: false,
      editPassword: false,
      editPhoneNumber: false,
      newPassword: '',
      password: '',
      passwordsMatch: false,
      user: this.props.authState.model,
      touchedFields: {}
    });
  }

  handleEditClick(e) {
    const editFlagName = e.target.parentElement.id;
    this.reset();
    this.setState({ [editFlagName]: true });
  }

  handlePasswordChange(event) {
    const { value } = event.target;
    this.setState({ password: value });
  }

  handleChange(event) {
    const { name, value } = event.target;
    const { user } = this.state;

    if (name === 'confirmPassword' || name === 'newPassword') {
      this.setState({ [name]: value });
      this.matchingPasswordCheck(name, value);
    } else {
      this.setState({
        user: {
          ...user,
          [name]: value
        }
      });
    }
  }

  handleBlur(e) {
    const touchedFields = Object.assign({}, this.state.touchedFields);
    touchedFields[e.target.name] = true;
    this.setState({ touchedFields });
  }

  matchingPasswordCheck(name, value) {
    const { newPassword, confirmPassword } = this.state;
    let passwordsMatch = false;

    if (name === 'confirmPassword') {
      passwordsMatch =
        newPassword === value && newPassword.length >= 8 && value.length >= 8;
    } else {
      passwordsMatch =
        value === confirmPassword &&
        newPassword.length >= 8 &&
        value.length >= 8;
    }

    this.setState({ passwordsMatch });
    return passwordsMatch;
  }

  savePersonalInfo(e) {
    e.preventDefault();

    const editFlagName = e.target.parentElement.id;
    const { user } = this.state;

    if (editFlagName === 'editPhoneNumber') {
      if (validatePhoneNumber(user.phoneNumber)) {
        this.props.userActions.update(user);
        this.reset();
      }
    } else {
      if (user.firstName.length >= 1 && user.lastName.length >= 1) {
        this.props.userActions.update(user);
        this.reset();
      }
    }
  }

  saveEmail(e) {
    e.preventDefault();

    const { user, password } = this.state;

    if (validateEmail(user.email) && password !== '' && password.length >= 8) {
      this.props.userActions.updateEmail({
        email: user.email,
        password
      });
      this.reset();
    } else {
      const touchedFields = Object.assign({}, this.state.touchedFields);
      touchedFields.password = true;
      this.setState({ touchedFields });
    }
  }

  savePassword(e) {
    e.preventDefault();

    const {
      password,
      passwordsMatch,
      newPassword,
      confirmPassword
    } = this.state;

    if (passwordsMatch && password !== '' && password.length >= 8) {
      this.props.userActions.changePassword({
        password,
        newPassword,
        confirmNewPassword: confirmPassword
      });
      this.reset();
    }
  }

  render() {
    const {
      user,
      editName,
      editEmail,
      editPhoneNumber,
      editPassword,
      newPassword,
      confirmPassword,
      password,
      passwordsMatch,
      touchedFields
    } = this.state;

    return (
      <ContactInfoContainer>
        Basic Contact Information
        <br />
        <Setting edit={editName} id="editName">
          Name
          <LinkButton
            onClick={this.handleEditClick}
            content="EDIT"
            right
            settings
          />
          <SettingCancelButton onClick={this.reset} />
          <SettingSaveButton onClick={this.savePersonalInfo} />
          <SettingReadonlyValue>
            {user.firstName} {user.lastName}
          </SettingReadonlyValue>
          <SettingEditableValue>
            <Spacing margin="-10px 0px 0px 0px">
              <Input
                type="text"
                name="firstName"
                id="firstName"
                placeholder="First Name"
                value={user.firstName}
                valid={user.firstName.length >= 1 && touchedFields.firstName}
                inValid={user.firstName.length < 1 && touchedFields.firstName}
                onChange={this.handleChange}
                onBlur={this.handleBlur}
              />
              <Input
                type="text"
                name="lastName"
                id="lastName"
                placeholder="Last Name"
                value={user.lastName}
                valid={user.lastName.length >= 1 && touchedFields.lastName}
                inValid={user.lastName.length < 1 && touchedFields.lastName}
                onChange={this.handleChange}
                onBlur={this.handleBlur}
              />
            </Spacing>
          </SettingEditableValue>
        </Setting>
        <Setting edit={editEmail} id="editEmail">
          Contact Email
          <LinkButton
            onClick={this.handleEditClick}
            content="EDIT"
            right
            settings
          />
          <SettingCancelButton onClick={this.reset} />
          <SettingSaveButton onClick={this.saveEmail} />
          <SettingReadonlyValue>{user.email}</SettingReadonlyValue>
          <SettingEditableValue>
            <Spacing margin="-10px 0px 0px 0px">
              <Input
                type="text"
                name="email"
                id="email"
                placeholder="Email"
                value={user.email}
                onChange={this.handleChange}
                valid={validateEmail(user.email) && touchedFields.email}
                inValid={!validateEmail(user.email) && touchedFields.email}
                onBlur={this.handleBlur}
              />
              <HelpBlockDiv
                type="alert-danger"
                show={!validateEmail(user.email) && touchedFields.email}
              >
                A valid Email is required
              </HelpBlockDiv>
              <Spacing margin="30px 0px 0px 0px">
                <Input
                  type="password"
                  name="password"
                  id="password"
                  placeholder="Password"
                  value={password}
                  onChange={this.handlePasswordChange}
                  valid={password.length >= 8}
                  inValid={password.length < 8 && touchedFields.password}
                  onBlur={this.handleBlur}
                />
                <HelpBlockDiv
                  type="alert-danger"
                  show={password.length < 8 && touchedFields.password}
                >
                  Password must be at least 8 characters
                </HelpBlockDiv>
              </Spacing>
            </Spacing>
          </SettingEditableValue>
        </Setting>
        <Setting edit={editPhoneNumber} id="editPhoneNumber">
          Phone Number
          <LinkButton
            onClick={this.handleEditClick}
            content="EDIT"
            right
            settings
          />
          <SettingCancelButton onClick={this.reset} />
          <SettingSaveButton onClick={this.savePersonalInfo} />
          <SettingReadonlyValue>{user.phoneNumber || '-'}</SettingReadonlyValue>
          <SettingEditableValue>
            <Spacing margin="-10px 0px 0px 0px">
              <Input
                type="text"
                name="phoneNumber"
                id="phoneNumber"
                placeholder="Phone Number"
                value={user.phoneNumber || ''}
                valid={
                  validatePhoneNumber(user.phoneNumber) &&
                  touchedFields.phoneNumber
                }
                inValid={
                  !validatePhoneNumber(user.phoneNumber) &&
                  touchedFields.phoneNumber
                }
                onChange={this.handleChange}
                onBlur={this.handleBlur}
              />
              <HelpBlockDiv
                type="alert-danger"
                show={
                  !validatePhoneNumber(user.phoneNumber) &&
                  touchedFields.phoneNumber
                }
              >
                Must be a valid U.S. Phone Number or Left Blank
              </HelpBlockDiv>
            </Spacing>
          </SettingEditableValue>
        </Setting>
        <Setting edit={editPassword} id="editPassword">
          Change Password
          <LinkButton
            onClick={this.handleEditClick}
            content="EDIT"
            right
            settings
          />
          <SettingCancelButton onClick={this.reset} />
          <SettingSaveButton onClick={this.savePassword} />
          <SettingReadonlyValue>********</SettingReadonlyValue>
          <SettingEditableValue>
            <Spacing margin="-10px 0px 0px 0px">
              <Input
                type="password"
                name="password"
                id="oldPassword"
                placeholder="Old Password"
                value={password}
                onChange={this.handlePasswordChange}
                valid={password.length >= 8}
                inValid={password.length < 8 && touchedFields.password}
                onBlur={this.handleBlur}
              />
              <HelpBlockDiv
                type="alert-danger"
                show={password.length < 8 && touchedFields.password}
              >
                Password must be at least 8 characters
              </HelpBlockDiv>
              <Spacing margin="30px 0px 0px 0px">
                <Input
                  type="password"
                  name="newPassword"
                  id="newPassword"
                  placeholder="New Password"
                  value={newPassword}
                  onChange={this.handleChange}
                  valid={
                    passwordsMatch &&
                    newPassword !== '' &&
                    touchedFields.confirmPassword
                  }
                  inValid={
                    (!passwordsMatch || newPassword === '') &&
                    touchedFields.confirmPassword
                  }
                />
                <Input
                  type="password"
                  name="confirmPassword"
                  id="confirmPassword"
                  placeholder="Confirm Password"
                  value={confirmPassword}
                  onChange={this.handleChange}
                  valid={
                    passwordsMatch &&
                    confirmPassword !== '' &&
                    touchedFields.confirmPassword
                  }
                  inValid={
                    (!passwordsMatch || confirmPassword === '') &&
                    touchedFields.confirmPassword
                  }
                  onBlur={this.handleBlur}
                />
                <HelpBlockDiv
                  type="alert-danger"
                  show={!passwordsMatch && touchedFields.confirmPassword}
                >
                  Passwords must match and be at least 8 characters
                </HelpBlockDiv>
              </Spacing>
            </Spacing>
          </SettingEditableValue>
        </Setting>
      </ContactInfoContainer>
    );
  }
}

ContactInfoSettings.propTypes = {
  userActions: PropTypes.shape(),
  authState: PropTypes.shape({
    model: PropTypes.shape()
  })
};

function mapStateToProps(state) {
  return {
    authState: state.auth
  };
}

function mapDispatchToProps(dispatch) {
  return {
    userActions: bindActionCreators(userActions, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(
  ContactInfoSettings
);
