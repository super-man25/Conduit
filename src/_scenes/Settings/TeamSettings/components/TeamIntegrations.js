// @flow

import React, { useState } from 'react';
import styled from 'styled-components';

import type { EDIntegration } from '_models';
import { Flex, FlexItem, H5, HR, S1, Integration, Box } from '_components';
import { IntegrationToggleAlertModal } from './IntegrationToggleAlertModal';

const TeamIntegrationsWrapper = styled(Flex)`
  flex-direction: column;
  justify-content: space-between;
`;

type ClientIntegration = {
  id: number,
  appId: string,
  name: string,
  version: string,
  createdAt: Date,
  modifiedAt: Date,
  isPrimary: boolean,
  isActive: boolean,
};

type Props = {
  primary: Array<ClientIntegration>,
  secondary: Array<ClientIntegration>,
  handleIntegrationToggle: (i: ClientIntegration, e: any) => void,
};

export const TeamIntegrations = (props: Props) => {
  const { primary, secondary, handleIntegrationToggle } = props;
  const [toggledIntegration: ?EDIntegration, setToggledIntegration] = useState(
    null
  );

  const onIntegrationChanged = (integration: EDIntegration) => {
    setToggledIntegration(integration);
  };

  const onCancel = () => setToggledIntegration(null);

  const onConfirmToggleIntegration = () => {
    if (!toggledIntegration) return;

    props.handleIntegrationToggle(
      toggledIntegration,
      !toggledIntegration.isActive
    );
    setToggledIntegration(null);
  };

  return (
    <TeamIntegrationsWrapper>
      <Box>
        <H5 type="secondary">Integrations</H5>
        {!!primary && (
          <React.Fragment>
            <S1 weight="300">
              <i>Primary Integration</i>
            </S1>
            <HR />
            <Flex flexWrap="wrap">
              {primary.map((i, index) => (
                <FlexItem flex="0 1 auto" margin="1rem" key={index}>
                  <Integration
                    {...i}
                    onChange={(e) => handleIntegrationToggle(i, e)}
                  />
                </FlexItem>
              ))}
            </Flex>
          </React.Fragment>
        )}
      </Box>
      {!!secondary && (
        <Box marginTop="4rem">
          <React.Fragment>
            <S1 weight="300">
              <i>Secondary Integrations</i>
            </S1>
            <HR />
            <Flex flexWrap="wrap">
              {secondary.map((i, index) => (
                <FlexItem
                  flex="0 1 auto"
                  margin="1rem"
                  height="245px"
                  key={index}
                >
                  <Integration
                    {...i}
                    onChange={(e) => onIntegrationChanged(i)}
                  />
                </FlexItem>
              ))}
            </Flex>
          </React.Fragment>
        </Box>
      )}
      {toggledIntegration && (
        <IntegrationToggleAlertModal
          onCancel={onCancel}
          onConfirm={onConfirmToggleIntegration}
          integration={toggledIntegration}
        />
      )}
    </TeamIntegrationsWrapper>
  );
};
