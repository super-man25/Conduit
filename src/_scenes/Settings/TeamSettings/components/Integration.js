// @flow
import React, { useState } from 'react';
import styled from 'styled-components';
import { useDispatch } from 'react-redux';

import { colors } from '_constants';
import {
  EDText,
  Button,
  Modal,
  PrimaryButton,
  SecondaryButton,
} from '_components';
import { readableDuration, sentenceCase } from '_helpers/string-utils';
import { actions as clientActions } from '_state/client';
import { SecondaryPricingRule } from './SecondaryPricingRule';

const Card = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background-color: white;
  width: 20rem;
  box-shadow: 0 4px 4px 0 ${colors.lightGray};
  padding: 25px;
  border: 1px solid ${colors.lightGray};
  border-radius: 6px;
  height: 100%;
`;

const Details = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const Img = styled.img`
  max-height: 5rem;
  object-fit: contain;
  object-position: left;
  width: 0;
  flex-grow: 1;
  margin-right: 50px;
`;

const DisableButton = styled(Button)`
  background-color: ${colors.red};
  color: white;
  padding: 3px;
  font-size: 10px;
  min-width: 0;
  max-width: 33%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  align-self: flex-start;

  &:hover {
    background-color: ${colors.red};
  }

  &:disabled {
    background-color: ${colors.lightGray};
    color: ${colors.gray};
  }
`;

const DisableIcon = styled.svg`
  color: inherit;
  fill: currentColor;
  height: 12px;
  width: 12px;
`;

const ConfirmationModal = styled(Modal)`
  background-color: white;
  padding: 25px;
  width: 500px;
  border: 1px solid ${colors.lightGray};
  border-radius: 6px;
`;

const ConfirmationText = styled.div``;

const ConfirmationButtons = styled.div`
  display: flex;
  justify-content: flex-end;
  margin-top: 50px;
`;

const CancelButton = styled(SecondaryButton)``;

const ContinueButton = styled(PrimaryButton)``;

type Props = {
  id: number,
  img?: String,
  modifiedAt: string,
  isActive: boolean,
  isPrimary: boolean,
  logoUrl?: string,
  name: string,
  percent?: number,
  constant?: number,
};

export const Integration = (props: Props) => {
  const { id, modifiedAt, isActive, logoUrl, name, isPrimary } = props;

  const [pendingConfirmation, setPendingConfirmation] = useState(false);

  const dispatch = useDispatch();
  const disableIntegration = () => {
    setPendingConfirmation(false);
    dispatch(clientActions.disableIntegration(id));
  };

  const closeModal = () => setPendingConfirmation(false);

  const allowDisabling = !isPrimary && isActive;

  return (
    <>
      <Card>
        <Details>
          <Img src={logoUrl} alt={name} />
          <DisableButton
            disabled={!allowDisabling}
            title="Disable Integration"
            onClick={allowDisabling ? () => setPendingConfirmation(true) : null}
          >
            <DisableIcon
              xmlns="http://www.w3.org/2000/svg"
              viewBox="0 0 1024 1024"
              version="1.1"
            >
              <path d="M508 990.4c-261.6 0-474.4-212-474.4-474.4S246.4 41.6 508 41.6s474.4 212 474.4 474.4S769.6 990.4 508 990.4zM508 136.8c-209.6 0-379.2 169.6-379.2 379.2 0 209.6 169.6 379.2 379.2 379.2s379.2-169.6 379.2-379.2C887.2 306.4 717.6 136.8 508 136.8zM697.6 563.2 318.4 563.2c-26.4 0-47.2-21.6-47.2-47.2 0-26.4 21.6-47.2 47.2-47.2l379.2 0c26.4 0 47.2 21.6 47.2 47.2C744.8 542.4 724 563.2 697.6 563.2z" />
            </DisableIcon>
          </DisableButton>
        </Details>
        {!isPrimary && <SecondaryPricingRule {...props} />}
        {!isPrimary && (
          <EDText size="small" type="tertiary" marginTop="15px">
            {sentenceCase(`last updated ${readableDuration(modifiedAt)} ago`)}
          </EDText>
        )}
      </Card>
      {pendingConfirmation && (
        <ConfirmationModal closeModal={closeModal}>
          <ConfirmationText>
            This will permanently delist all tickets from {name}, would you like
            to continue? To re-enable this integration, contact
            product@eventdynamic.com
          </ConfirmationText>
          <ConfirmationButtons>
            <CancelButton onClick={closeModal}>Cancel</CancelButton>
            <ContinueButton onClick={disableIntegration}>
              Disable Integration
            </ContinueButton>
          </ConfirmationButtons>
        </ConfirmationModal>
      )}
    </>
  );
};
