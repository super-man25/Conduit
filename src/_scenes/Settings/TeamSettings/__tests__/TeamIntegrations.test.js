import React from 'react';
import { shallow } from 'enzyme';
import { TeamIntegrations } from '../components/TeamIntegrations';

describe('<TeamIntegrations />', () => {
  const props = {
    primary: [
      {
        name: 'tickets.com',
        isPrimary: true,
        isActive: true,
      },
    ],
    secondary: [
      {
        name: 'seatgeek.com',
        isPrimary: false,
        isActive: true,
      },
    ],
    handleIntegrationToggle: jest.fn(),
    handleSecondaryPriceRuleUpdate: jest.fn(),
  };

  it('render correctly', () => {
    const wrapper = shallow(<TeamIntegrations {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
