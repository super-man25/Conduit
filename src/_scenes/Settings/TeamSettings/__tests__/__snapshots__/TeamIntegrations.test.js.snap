// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamIntegrations /> render correctly 1`] = `ShallowWrapper {}`;
